AppCenterAnalytics/AppCenterAnalytics/TransmissionTarget/MSPropertyConfigurator.m

```diff
@@ -46,6 +47,26 @@ - (void)setAppLocale:(NSString *)appLocale {
   _appLocale = appLocale;
 }
 
+- (void)setAppUserId:(NSString *)appUserId {
+  @synchronized([MSAnalytics sharedInstance]) {
```

Comment:
Any reason for synchronization? If we need this then we need to synchronize other methods.

Reply:
The check is not atomic with the update (if). We also do that in other existing methods that have more than just a variable assignment.

Reply:
Also it creates the static cached regex only once